### PR TITLE
fsck: list broken files

### LINKED ISSUE #949: Find out broken files in `fsck`
When doing `juicefs fsck`, we can scan out lost objects, but users probably want to know the broken files (paths) containing these objects.

### PR DESCRIPTION
**Pull request by SandyXSD**
close #949

### CODE REVIEW

**Comment by SandyXSD** [comment]
TODO: The GetPath may not work if the first link of a file is removed (attr.Parent is not properly maintained)

File: cmd/fsck.go [hunk 1]
```diff
@@ -127,24 +132,32 @@ func fsck(ctx *cli.Context) error {
...
+					sz = int(s.Size) - int(i)*chunkConf.BlockSize
+				}
+				key := fmt.Sprintf("%d_%d_%d", s.Chunkid, i, sz)
+				if _, ok := blocks[key]; !ok {
+					if _, err := blob.Head(key); err != nil {
+						if _, ok := brokens[inode]; !ok {
+							if p, st := m.GetPath(meta.Background, inode); st == 0 {
+								brokens[inode] = p
+							} else {
+								logger.Warnf("getpath of inode %d: %s", inode, st)
```
Comment (davies) [inline cmd/fsck.go:154]: use `inode:%d` as the path, otherwise it will be empty string

File: pkg/meta/tkv.go [hunk 2]
```diff
@@ -2357,6 +2357,39 @@ func (r *kvMeta) ListSlices(ctx Context, slices *[]Slice, delete bool, showProgr
...
+		}
+		for key := range vals {
+			names = append(names, key[len(m.entryKey(inode, "")):])
+			break
+		}
+		inode = attr.Parent
+	}
+	names = append(names, "") // add root
+
+	for i, j := 0, len(names)-1; i < j; i, j = i+1, j-1 { // reverse
```
Comment (davies) [inline pkg/meta/tkv.go:2387]: use a function for this

Comment (davies) [reply pkg/meta/tkv.go:2387]:
```
return "/" + strings.Join(reverse(names), "/"), 0
```

File: pkg/meta/interface.go [hunk 3]
```diff
@@ -307,7 +307,9 @@ type Meta interface {
 	// Compact all the chunks by merge small slices together
 	CompactAll(ctx Context) syscall.Errno
 	// ListSlices returns all slices used by all files.
-	ListSlices(ctx Context, slices *[]Slice, delete bool, showProgress func()) syscall.Errno
+	ListSlices(ctx Context, slices map[Ino][]Slice, delete bool, showProgress func()) syscall.Errno
+	// GetPath returns full path of an inode; a random one is picked if it has multiple hard links
+	GetPath(ctx Context, inode Ino) (string, syscall.Errno)
```
Comment (davies) [inline pkg/meta/interface.go:312]: nit: GetPath could be implemented use  `Lookup(ctx, inode, "..")` and `Readdir(ctx, inode, 0)` without introducing a new API.

**Review by davies — CHANGES_REQUESTED** [review]
This could be simplified.

File: pkg/meta/interface.go (hunk 3 above)
Comment (davies) [reply pkg/meta/interface.go:312]: But `Lookup(ctx, inode, "..")` does not work on files ...

File: cmd/fsck.go (hunk 1 above)
Comment (SandyXSD) [reply cmd/fsck.go:154]: It prints `$inode: $path` for each file, so maybe use the error message to replace path would be more comprehensive. Like:
```
juicefs[24127] <FATAL>: 3 objects are lost (24 bytes), 3 broken files:
        INODE: PATH
          325: /d3/f33
        32181: /fff
        32182: no such file or directory
```

File: pkg/meta/interface.go (hunk 3 above)
Comment (SandyXSD) [reply pkg/meta/interface.go]: Just use `GetAttr` for now

**Review by davies — APPROVED** [review]

**Pull request merged**